src/MassTransit/JobService/JobTypeStateMachine.cs

```diff
@@ -129,6 +129,20 @@ select new
             if (nextInstance == null)
                 return false;
 
+            var jobPriorityBucket = context.Message.JobPriority switch
+            {
+                JobPriority.Low => 0.5,
+                JobPriority.BelowNormal => 0.7,
+                JobPriority.Normal => 0.8,
+                JobPriority.AboveNormal => 0.9,
+                _ => 1.0
+            };
+
+            if (nextInstance.InstanceCount >= Math.Ceiling(concurrentJobLimit * jobPriorityBucket))
```

Comment:
Could `concurrentJobLimit` be 0? In which case, how does that affect this? It seems like it defaults to `1`. I guess if it is set to 0, then nothing should run.

Reply:
I don't think anything prevents you from setting 0 no, but yeah nothing would ever run. It would get rejected before this, on line 120. Probably won't bother adding a guard against that. 

There could be an argument in ensuring that `Math.Ceiling(concurrentJobLimit * jobPriorityBucket)` always evaluates to distinct values, for different priorities. For example, `AboveNormal` priority would behave like `High` priority, if `concurrentJobLimit < 10` (`Math.Ceiling(9 * 0.9) = 9` and `Math.Ceiling(9 * 1) = 9`). 
Though I couldn't immediately come up with a way of doing it, so I thought: ah, whatever, it's just a "best effort" anyways, and we are using concurrent job limits above 10.

Reply:
Hmm true. I guess you'd need to normalize the other side (the `concurrentJobLimit`/`InstanceCount`), so it calculates higher as expected each time. Can't think of a way how to at the moment mind :\

Reply:
I think I'll call it good enough for us at the moment, but it is this specific part of the code that I don't think is suitable for pushing upstream. It seems a bit of an unintuitive/magical approach